pytest_localserver/https.py

```diff
@@ -111,13 +113,41 @@ class SecureContentServer(ContentServer):
     .. _pyOpenSSH: https://launchpad.net/pyopenssl
     """
 
-    def __init__(self, host="localhost", port=0, key=DEFAULT_CERTIFICATE, cert=DEFAULT_CERTIFICATE):
+    def __init__(self, host="localhost", port=0, key=DEFAULT_KEY, cert=DEFAULT_CERTIFICATE):
         """
         :param key: location of file containing the server private key.
         :param cert: location of file containing server certificate.
         """
 
-        super().__init__(host, port, ssl_context=(key, cert))
+        super().__init__(host, port, ssl_context=(cert, key))
+        self._cert = cert
+        
+    @property
+    def url(self):
+        # IMPORTANT:
+        # The supplied certificate is bound exclusively to "localhost" and does not include "127.0.0.1" in its Subject Alternative Name (SAN) field.
+        # As a result, validation against "127.0.0.1" will fail, even if the server's certificate is added to the trust chain.
+        # Current SAN configuration:
+        #  DNS:localhost
+        #
+        # To ensure successful validation when accessing the server via IP address, the certificate must be reissued with "127.0.0.1" included in the SAN field.
```

Comment:
Honestly, I believe that not having to create the certificate oneself is a major advantage compared to `pytest-httpserver`. I think reissuing the certificate in the future would be a better approach.

Still it's your call as maintainer.

Reply:
Oh interesting, I didn't realize pytest-httpserver doesn't provide a default certificate.

Anyway, I wouldn't mind adding `127.0.0.1` to the SAN field in a separate PR (or I guess it could even be a separate commit in this PR, though I generally prefer splitting different features into different PRs). It's just that since we're not doing that yet, I thought it would be best to make the documentation reflect the current state of things for now.

Reply:
> Oh interesting, I didn't realize pytest-httpserver doesn't provide a default certificate.
> 

No it doesn't however, integrating it using the suggested method [Running an HTTPS server](https://pytest-httpserver.readthedocs.io/en/latest/howto.html#running-an-https-server) is fairly stright forward. Nonetheless, the `pytest-httpserver` method presents several challenges for my specific application, particularly in testing various scenarios involving self-signed certificates (therefore I have choosen `pytest-localserver`).

> Anyway, I wouldn't mind adding `127.0.0.1` to the SAN field in a separate PR (or I guess it could even be a separate commit in this PR... I personally prefer splitting different features into different PRs, but it's not important enough that I'd really insist on it here). I was just thinking that since we're not doing that yet, I thought it would be best to make the documentation reflect the current state of things for now.
> 
> Honestly, it wouldn't be too bad to have one commit that regenerates the certificate with `127.0.0.1` in the SAN and also splits it and the key into separate files. It is sort of bending the principle of making each commit focus on one thing, but those two changes are closely related enough that it seems like a reasonable exception in my opinion. 🤷 So if you want to do that in this PR, feel free to do so, I'm on board.

After giving this some more thinking, I believe that adding `127.0.0.1` to the SAN (Subject Alternative Name) field would be the most straightforward solution. Here's why:

1. It allows us to utilize the URL component from the parent class directly.
2. The URL will behave as expected, closely mirroring the behavior of the HTTP version.
3. Using `localhost` can lead to certain complications, as outlined in the video [Don't Use localhost (intermediate) - Anthony Explains](https://www.youtube.com/watch?v=98SYTvNw1kw).

So if you on board let's do it this way.

Reply:
Sounds good, we're agreed on adding `127.0.0.1` to the SAN field :+1:

As for the hostname, I'd rather stick with using `localhost` consistently throughout. The complications mentioned in that video are relevant in fairly niche situations, not the typical use case for this package. But we don't need to make that change as part of this PR. So I don't think any further changes are needed on this point.